test/util/general.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Generate a randomized hex string of a specific length
+ * @param length The length of the hex string
+ */
+export const generateHexString = (length: number): string => {
+    return (
+        [...Array(length)]
+            .map(() => Math.floor(Math.random() * 16).toString(16))
+            .join("")
+    );
+};
+
+/**
+ * Returns a random thing sampled from a list of things
+ * @param data - the array of things to sample from
+ */
+export const sample = <Type>(data: Type[]): Type => {
+    const sampleIdx = randInt(data.length - 1)
```

Comment:
randInt is exclusive of the max.

```suggestion
    const sampleIdx = randInt(data.length)
```